tests/test_runner/test_runner.py

```diff
@@ -884,6 +885,32 @@ def forward(self, *args, **kwargs):
             with self.assertRaises(ValueError):
                 Runner.from_cfg(cfg)
 
+    def test_distinguish_experiment(self):
+        self.temp_dir = 'work_dirs'
+        cfg = copy.deepcopy(self.epoch_based_cfg)
+        cfg.work_dir = None
+
+        # When do not auto distinguish experiments
+        # work_dir is set to work_dirs/experiment_name by default.
+        cfg.auto_distinguish_experiment = False
+        cfg.experiment_name = 'test_distinguish_experiment_false'
+        runner = Runner.from_cfg(cfg)
+        self.assertTrue(
+            runner.work_dir.endswith(
+                os.path.join('work_dirs',
+                             'test_distinguish_experiment_false')))
+
+        # When do not auto distinguish experiments
+        # work_dir is set to work_dirs/experiment_name/timestamp
+        # by default.
+        cfg.auto_distinguish_experiment = True
+        cfg.experiment_name = 'test_distinguish_experiment_true'
+        runner = Runner.from_cfg(cfg)
+        self.assertTrue(
+            runner.work_dir.endswith(
+                os.path.join('work_dirs', 'test_distinguish_experiment_true',
+                             runner.timestamp)))
```

Comment:
This test function can be merged into 
https://github.com/open-mmlab/mmengine/blob/601db12d38245140775ed189fcc3b4815843b567/tests/test_runner/test_runner.py#L420

Reply:
could we go to the `self.tempdir` by `os.chdir()`, and then build the runner with `None` `work_dir`？Finally we can go back to the origin directory.

Reply:
Got it.